presto-main/src/main/java/io/prestosql/operator/TableScanWorkProcessorOperator.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.memory.context.AggregatedMemoryContext;
+import io.prestosql.memory.context.LocalMemoryContext;
+import io.prestosql.memory.context.MemoryTrackingContext;
+import io.prestosql.metadata.Split;
+import io.prestosql.metadata.TableHandle;
+import io.prestosql.operator.WorkProcessor.ProcessState;
+import io.prestosql.operator.WorkProcessor.TransformationState;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.connector.ConnectorPageSource;
+import io.prestosql.spi.connector.UpdatablePageSource;
+import io.prestosql.split.PageSourceProvider;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import static io.airlift.concurrent.MoreFutures.toListenableFuture;
+import static java.util.Objects.requireNonNull;
+
+public class TableScanWorkProcessorOperator
+        implements WorkProcessorSourceOperator
+{
+    private final WorkProcessor<Page> pages;
+
+    private ConnectorPageSource source;
```

Comment:
In stead of having this field, I suggest we move it to `SplitToPages` and have a field for `SplitToPages`.  Then then add `destroy()` and `getUpdatablePageSourceSupplier` methods on `SplitToPages`.  This way `SplitToPages` can be static and encapsulate it's full state.. which I hope will make this easier to read.